src/App.tsx

```diff
@@ -1,16 +1,25 @@
+import { FC, useState } from 'react';
+
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
+
+export const App: FC = () => {
+  const [movies, setMovies] = useState<Movie[]>(moviesFromServer);
+
+  const addMovie = (movie: Movie) => {
+    setMovies(currentMovies => [...currentMovies, movie]);
+  };
 
-export const App = () => {
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={movies} />
       </div>
       <div className="sidebar">
-        <NewMovie /* onAdd={(movie) => {}} */ />
+        <NewMovie onAdd={addMovie} />
```

Comment:
```suggestion
        <NewMovie onAdd={addMovie} key={count} />
```
Move counter logic here. When you'll increase count your component will be created from scratch with default values and no errors.